app/controllers/concerns/search.rb

```diff
@@ -7,8 +7,16 @@ module Search
   included do
     def scope_search
       result = policy_scope(resource)
+      escape_references_search
       result = result.search_for(params[:search]) if params[:search].present?
       result.paginate(page: pagination_offset, per_page: pagination_limit)
     end
+
+    def escape_references_search
+      return unless params[:search]&.match(/reference=/)
+
+      search_term = CGI.escape(params[:search].split('=')[1])
```

Comment:
CGI escape is overkill. We can just surround the search term in double quotes.
```suggestion
      search_term = "\"#{params[:search].split('=')[1]}\""
```